File: mobile/lib/models/product.dart
class Product {
    String id;
    String name;
    String description;
    String price;
    String imgUrlColorOne;
    String imgUrlColorTwo;
    DateTime createdAt;
    int v;

    Product({
        this.id,
        this.name,
        this.description,
        this.price,
        this.imgUrlColorOne,
        this.imgUrlColorTwo,
        this.createdAt,
        this.v,
    });

    factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json["_id"],
        name: json["name"],
        description: json["description"],
        price: json["price"],
        imgUrlColorOne: json["imgUrlColorOne"],
        imgUrlColorTwo: json["imgUrlColorTwo"],
        createdAt: DateTime.parse(json["createdAt"]),
        v: json["__v"],
    );
}

File: backend/src/models/Product.js
const mongoose = require('mongoose');
const moongosePaginate = require('mongoose-paginate');

const ProductSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        required: true,
    },
    price: {
        type: String,
        required: true,
    },
    imgUrlColorOne: {
        type: String,
        required: true,
    },
    imgUrlColorTwo:{
        type: String,
        required: true,
    },
    createdAt:{
        type: Date,
        default: Date.now,
    }
});

ProductSchema.plugin(moongosePaginate);

module.exports = mongoose.model('Product',ProductSchema);